devito/ops/node_factory.py

```diff
@@ -0,0 +1,63 @@
+from devito import Dimension, TimeFunction
+from devito.ops.utils import namespace
+from devito.symbolics import Macro, split_affine
+from devito.types import Indexed, Array
+
+
+class OPSNodeFactory():
+    """
+    Generate OPS nodes for building an OPS expression.
+
+    Examples
+    --------
+    >>> a = OPSNodeFactory()
+    >>> a.new_symbol('symbol_name')
+    symbol_name
+    """
+
+    def __init__(self):
+        self.ops_args = {}
+
+    def new_ops_arg(self, indexed):
+        """
+        Create an :class:`Indexed` node using OPS representation.
+        If the pair ops argument name and time dimension were alredy created,
```

Comment:
"If the pair ``(argname, time_dimension)`` ...
feels like this needs some better wording. It's not entirely clear why you do so, for example (well, it is clear to me, but probably not to the newcomer...)